e2e/test/scenarios/models/reproductions.cy.spec.ts

```diff
@@ -668,3 +668,90 @@ describe("issue 39749", () => {
     cy.findByLabelText("Description").should("have.text", "B");
   });
 });
+
+describe("issue 25885", () => {
+  const mbqlModelDetails: StructuredQuestionDetails = {
+    type: "model",
+    query: {
+      "source-table": ORDERS_ID,
+      fields: [["field", ORDERS.ID, { "base-type": "type/BigInteger" }]],
+      joins: [
+        {
+          fields: [
+            [
+              "field",
+              ORDERS.ID,
+              { "base-type": "type/BigInteger", "join-alias": "Orders" },
+            ],
+          ],
+          strategy: "left-join",
+          alias: "Orders",
+          condition: [
+            "=",
+            ["field", ORDERS.ID, { "base-type": "type/BigInteger" }],
+            [
+              "field",
+              ORDERS.ID,
+              { "base-type": "type/BigInteger", "join-alias": "Orders" },
+            ],
+          ],
+          "source-table": ORDERS_ID,
+        },
+        {
+          fields: [
+            [
+              "field",
+              ORDERS.ID,
+              { "base-type": "type/BigInteger", "join-alias": "Orders_2" },
+            ],
+          ],
+          strategy: "left-join",
+          alias: "Orders_2",
+          condition: [
+            "=",
+            ["field", ORDERS.ID, { "base-type": "type/BigInteger" }],
+            [
+              "field",
+              ORDERS.ID,
+              { "base-type": "type/BigInteger", "join-alias": "Orders_2" },
+            ],
+          ],
+          "source-table": ORDERS_ID,
+        },
+      ],
+    },
+  };
+
+  beforeEach(() => {
+    restore();
+    cy.signInAsNormalUser();
+    cy.intercept("PUT", "/api/card/*").as("updateModel");
+  });
+
+  function setColumnName(oldName: string, newName: string) {
+    tableHeaderClick(oldName);
+    cy.findByLabelText("Display name")
+      .should("have.value", oldName)
+      .clear()
+      .type(newName);
+  }
+
+  function verifyColumnName(name: string) {
+    tableHeaderClick(name);
+    cy.findByLabelText("Display name").should("have.value", name);
+  }
+
+  it("should allow to edit metadata for mbql models with self joins columns (metabase#25885)", () => {
```

Comment:
Just to make it explicit, I'm not closing the issue as metadata reconciliation is broken on the BE right now.